### PR TITLE
Builders bindings tests for python&java

### PR DESCRIPTION
**Pull request by l4l**
Rebased #764 

So far in this pr:
- Java builders tests
- Python builders tests
- gradle integration for java

### CODE REVIEW

File: test/module/shared_model/bindings/BuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class BuilderTest {
+    static {
+        try {
+            System.loadLibrary("iroha");
+        } catch (UnsatisfiedLinkError e) {
+            System.err.println("Native code library failed to load. \n" + e);
```
Comment (muratovv) [inline test/module/shared_model/bindings/BuilderTest.java]: Is it a part of the test? I mean if the iroha lib not loaded test should be failed.

File: test/module/shared_model/bindings/BuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            System.exit(1);
+        }
+    }
+
+    private Keypair keys;
+    private ModelBuilder builder;
+
+    ModelBuilder base() {
+        return new ModelBuilder().txCounter(BigInteger.valueOf(123))
+                .createdTime(BigInteger.valueOf(System.currentTimeMillis()))
```
Comment (muratovv) [inline test/module/shared_model/bindings/BuilderTest.java]: Maybe we need to think about the possibility of setting counter automatically from the native library.

File: test/module/shared_model/bindings/BuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+    @Test
+    void emptyTx() {
+        ModelBuilder builder = new ModelBuilder();
+        assertThrows(IllegalArgumentException.class, builder::build);
+    }
+
+    @Test
+    void txWithoutCommand() {
+//        broken for now
```
Comment (muratovv) [inline test/module/shared_model/bindings/BuilderTest.java]: What does it mean? What there is need to do for fix it?

File: test/module/shared_model/bindings/builder-test.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  def setUp(self):
+    self.keys = iroha.ModelCrypto().generateKeypair()
+    self.builder = self.generate_base()
+
+  def set_add_peer(self):
+    self.builder.addPeer("123.123.123.123", self.keys.publicKey())
+
+  def test_tx_without_command(self):
+    return #broken for now
```
Comment (muratovv) [inline test/module/shared_model/bindings/builder-test.py]: Same issue as with Java.

File: test/module/shared_model/bindings/builder-test.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import iroha
```
Comment (muratovv) [inline test/module/shared_model/bindings/builder-test.py]: Is there possible that Iroha lib failed on initialization? If yes, this cases handled by the test?

File: test/module/shared_model/bindings/BuilderTest.java (hunk 1 above)
Comment (l4l) [reply test/module/shared_model/bindings/BuilderTest.java]: There's exit(1) below that will fail test anyway. I'll replace it with an explicit assertion

File: test/module/shared_model/bindings/BuilderTest.java (hunk 3 above)
Comment (l4l) [reply test/module/shared_model/bindings/BuilderTest.java]: For some reason current shared_model doesn't allow to build tx w/o any command. That's a point for discussion primarily. If it's fine I can just delete this test

File: test/module/shared_model/bindings/builder-test.py (hunk 5 above)
Comment (l4l) [reply test/module/shared_model/bindings/builder-test.py]: Then it'll be exception with smth like "sorry cannot find this stuff" and tests fails as long as it doesn't be handled

File: test/module/shared_model/bindings/BuilderTest.java (hunk 3 above)
Comment (luckychess) [reply test/module/shared_model/bindings/BuilderTest.java]: Some time ago we decided that tx without commands is invalid (in chat discussion I believe) and I did a check in validator for this case.

Comment (l4l) [reply test/module/shared_model/bindings/BuilderTest.java]: Ok, I'll delete it then

**Pull request closed without merging**